src/System.Memory/src/System/NUint.cs

```diff
@@ -29,6 +29,68 @@ internal struct NUInt
             unsafe { return (sizeof(IntPtr) == 4) ? new NUInt(((uint)left._value) * (uint)right) : new NUInt(((ulong)left._value) * (uint)right); }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static NUInt operator +(NUInt left, int right)
+        {
+            unsafe { return (sizeof(IntPtr) == 4) ? new NUInt(((uint)left._value) + (uint)right) : new NUInt(((ulong)left._value) + (uint)right); }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static NUInt operator -(NUInt left, int right)
+        {
+            unsafe { return (sizeof(IntPtr) == 4) ? new NUInt(((uint)left._value) - (uint)right) : new NUInt(((ulong)left._value) - (uint)right); }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static NUInt operator *(NUInt left, NUInt right)
+        {
+            unsafe { return (sizeof(IntPtr) == 4) ? new NUInt(((uint)left._value) * (uint)right._value) : new NUInt(((ulong)left._value) * (uint)right._value); }
```

Comment:
The rightmost cast of `right` to `NUint` will cause value loss. Same with other operators that take a `NUint right`